library/aria.c

```diff
@@ -960,15 +959,45 @@ int mbedtls_aria_self_test( int verbose )
         /* test ECB encryption */
         if( verbose )
             mbedtls_printf( "  ARIA-ECB-%d (enc): ", 128 + 64 * i );
-        mbedtls_aria_setkey_enc( &ctx, aria_test1_ecb_key, 128 + 64 * i );
-        mbedtls_aria_crypt_ecb( &ctx, aria_test1_ecb_pt, blk );
+        ret = mbedtls_aria_setkey_enc( &ctx, aria_test1_ecb_key, 128 + 64 * i );
+        if( ret == MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED )
+        {
+            if( verbose != 0 )
+                mbedtls_printf( "skipped\n" );
+            continue;
```

Comment:
We are adding a lot of branches to handle FEATURE_UNSUPPORTED from many places. How will we test this? Our code coverage is going to suffer a lot unless we come up with and write tests to hit each of these new branches. This can also help us find bugs if we don't properly handle FEATURE_UNSUPPORTED errors somewhere.

Reply:
I agree, but as mentioned, since we allow this eror to be returned in every function, I don't see another option.
My suggestion was to have "dummy targets" that will return this error in different functions

Reply:
Yes, we need dummy targets (more commonly known as test mocks) in order to complete this feature.